src/java.base/share/classes/java/lang/String.java

```diff
@@ -1284,13 +1284,18 @@ private static byte[] encodeUTF8(byte coder, byte[] val, boolean doReplace) {
             return encodeUTF8_UTF16(val, doReplace);
         }
 
-        if (!StringCoding.hasNegatives(val, 0, val.length)) {
+        int positives = StringCoding.countPositives(val, 0, val.length);
+        if (positives == val.length) {
             return val.clone();
         }
 
-        int dp = 0;
         byte[] dst = StringUTF16.newBytesFor(val.length);
-        for (byte c : val) {
+        if (positives > 0) {
```

Comment:
I wonder if for very short strings, the overhead of the setup for `arraycopy` is worth it.
It might be interesting to raise the test here to 4 or 8 for the array copy  and otherwise start the loop at 0 and see if the performance difference is detectable.